test/e2e/providers/azure/azure.go

```diff
@@ -119,11 +124,218 @@ func (p *Provider) GetType() config.PlatformType {
+	}
+	storageClassName := "azurefile-csi"
+	pvcSpec := core.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			GenerateName: "e2e" + "-",
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			AccessModes: []core.PersistentVolumeAccessMode{core.ReadWriteMany},
+			Resources: core.ResourceRequirements{
+				Requests: core.ResourceList{core.ResourceStorage: resource.MustParse("2Gi")},
+			},
+			StorageClassName: &storageClassName,
+		},
+	}
+	return c.CoreV1().PersistentVolumeClaims(namespace).Create(context.TODO(), &pvcSpec, meta.CreateOptions{})
+}
+
+// ensureWindowsCSIDaemonSet deploys the Windows CSI driver DaemonSet if it doesn't already exist
+func (p *Provider) ensureWindowsCSIDaemonSet(client client.Interface) error {
+	dsName := "azure-file-csi-driver-node-windows"
+	directoryType := core.HostPathDirectory
+	directoryOrCreate := core.HostPathDirectoryOrCreate
+	ds := apps.DaemonSet{
+		ObjectMeta: meta.ObjectMeta{
+			Name: dsName,
+		},
+		Spec: apps.DaemonSetSpec{
+			Selector: &meta.LabelSelector{MatchLabels: map[string]string{"app": dsName}},
+			Template: core.PodTemplateSpec{
+				ObjectMeta: meta.ObjectMeta{
+					Labels: map[string]string{"app": dsName},
+				},
+				Spec: core.PodSpec{
+					PriorityClassName: "system-node-critical",
+					NodeSelector:      map[string]string{core.LabelOSStable: "windows"},
+					// Use the controller-sa as the node-sa doesn't have GET secrets permissions required for Windows
+					ServiceAccountName: "azure-file-csi-driver-controller-sa",
+					OS:                 &core.PodOS{Name: core.Windows},
+					Tolerations: []core.Toleration{
+						{
+							Key:    "os",
+							Value:  "Windows",
+							Effect: core.TaintEffectNoSchedule,
+						},
+					},
+					Containers: []core.Container{
+						{
+							Name:  "node-driver-registrar",
+							Image: "mcr.microsoft.com/oss/kubernetes-csi/csi-node-driver-registrar:v2.8.0",
+							Args:  []string{"--v=2", "--csi-address=$(CSI_ENDPOINT)", "-kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)"},
+							Env: []core.EnvVar{
+								{
+									Name:  "CSI_ENDPOINT",
+									Value: `unix://C:\\csi\\csi.sock`,
+								},
+								{
+									Name:  "DRIVER_REG_SOCK_PATH",
+									Value: `C:\\var\\lib\\kubelet\\plugins\\file.csi.azure.com\\csi.sock`,
+								},
+								{
+									Name: "KUBE_NODE_NAME",
+									ValueFrom: &core.EnvVarSource{
+										FieldRef: &core.ObjectFieldSelector{
+											FieldPath: "spec.nodeName",
+										},
+									},
+								},
+							},
+							VolumeMounts: []core.VolumeMount{
+								{
+									Name:      "kubelet-dir",
+									MountPath: "/var/lib/kubelet",
+								},
+								{
+									Name:      "plugin-dir",
+									MountPath: "/csi",
+								},
+								{
+									Name:      "registration-dir",
+									MountPath: "/registration",
+								},
+							},
+						},
+						{
+							Name:  "azurefile",
+							Image: "mcr.microsoft.com/k8s/csi/azurefile-csi:latest",
```

Comment:
Should we worry about matching compatible image versions of the `azurefile-csi` and `csi-node-driver-registrar`?

Reply:
This is following the documentation provided https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/deploy